### DISCUSSION TITLE
Add module for PowerDNS zone management

### DISCUSSION THREAD

**Discussion by AliakseiKorneu** [saltstack/salt#67349] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Need a simple tool for managing my PowerDNS instance: add zones, update zones records, delete old zones, and so on. 

**Describe the solution you'd like**
It would be great to have state and execution modules for salt for managing PowerDNS zones and records to automate such a process

**Describe alternatives you've considered**
As an alternative, it's possible to use PowerDNS API or UI but it's not convenient for large amount of zones and records.

**Comment by welcome** [comment]
Hi there! Welcome to the Salt Community! Thank you for making your first contribution. We have a lengthy process for issues and PRs. Someone from the Core Team will follow up as soon as possible. In the meantime, here’s some information that may help as you continue your Salt journey.
Please be sure to review our [Code of Conduct](https://github.com/saltstack/salt/blob/master/CODE_OF_CONDUCT.md). Also, check out some of our community resources including:

  - [Community Wiki](https://github.com/saltstack/community/wiki)
  - [Salt’s Contributor Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html)
  - [Join our Community Slack](https://saltstackcommunity.herokuapp.com/)
  - [IRC on Freenode](https://webchat.freenode.net/#salt)
  - [SaltStack YouTube channel](https://www.youtube.com/user/SaltStack)
  - [SaltStackInc Twitch channel](https://www.twitch.tv/saltstackinc)

There are lots of ways to get involved in our community. Every month, there are around a dozen opportunities to meet with other contributors and the Salt Core team and collaborate in real time. The best way to keep track is by subscribing to the Salt Community Events Calendar.
If you have additional questions, email us at core@saltstack.com. We’re glad you’ve joined our community and look forward to doing awesome things with you!

**Comment by s0undt3ch** [comment]
If you have the knowledge, I'd suggest to start writing a salt module, and state module, to achieve what you're after.

I'm approving the feature request because it is a valid feature request, although, it does not mean someone from the core team will start working on the feature. If however, you, or someone from the Salt community, steps up to do the work, then this could be implemented sooner.
